### ISSUE TITLE
getter-return: should permit generators

### ISSUE THREAD

**Issue by mfulton26** [denoland/deno_lint#1072]
### Lint Name

getter-return

### Code Snippet

```ts
const x = Symbol();

Object.defineProperty(Number.prototype, x, {
  *get() {
    for (let n = 0; n < this; n++) yield n;
  },
});

declare global {
  interface Number {
    [x]: IterableIterator<number>;
  }
}

Array.from(5[x], (i) => i + 1);
// => [ 1, 2, 3, 4, 5 ]

```

### Expected Result

no lint warnings

### Actual Result

```warning
Expected to return a value in 'get'.
Return a value from the getter function
```

### Additional Info

A getter within the context of `Object.defineProperty`, `Object.defineProperties`, or `Reflect.defineProperty` should be able to be defined using a generator without specifying an explicit return value.

### Version

deno 1.25.0 (release, aarch64-apple-darwin)
v8 10.6.194.5
typescript 4.7.4

**Comment by hirasawayuki** [comment]
@magurotuna 
If `is_generator` of `Function` is true, skipping the check for the getter_return rule seems to solve this problem.
Should I also check for the existence of `YieldExpr`?

**code sinipet**
```ts
Object.defineProperty(Number.prototype, x, {
  *get() {
    for (let n = 0; n < this; n++) yield n;
  },
});
```
**ast**
```rust
function: Function {
  params: [],
  decorators: [],
  span: Span {
  ...
  },
  body: Some(
  ...
  ),
  is_generator: true,  // ← this
  is_async: false,
  type_params: None,
  return_type: None,
}
```

**Comment by magurotuna** [comment]
@hirasawayuki Yes, that approach sounds fine. Generators are beyond the scope of this rule, so there's no need to check `yield` in this rule in my opinion. In fact, we have another rule called `require-yield` that checks if generators have `yield` statements in there. It should be nice to leave the check for `yield` to this rule.
